src/lib.rs

```diff
@@ -111,24 +103,18 @@ where
     where
         P: GetNextMut<NextPtr = Option<P>>,
     {
-        unsafe {
-            let next = value.get_next() as *mut Option<P>;
-            let raw = value.into_raw();
-            // Iff next was set to Some(P) we want to
-            // assert that it was droppeds
-            drop(ptr::read(next));
-            loop {
-                let pcurrent = self.inner.load(load_order);
-                let current = if pcurrent.is_null() {
-                    None
-                } else {
-                    Some(FromRawPtr::from_raw(pcurrent))
-                };
-                ptr::write(next, current);
-                let last = self.inner.compare_and_swap(pcurrent, raw, cas_order);
-                if last == pcurrent {
-                    return last.is_null();
-                }
+        let next = value.get_next() as *mut Option<P>;
+        let raw = value.into_raw();
+        // Iff next was set to Some(P) we want to
+        // assert that it was droppeds
+        drop(unsafe { ptr::read(next) });
```

Comment:
https://github.com/slide-rs/atom/pull/6#discussion_r187780689

Reply:
@torkleyy I think this function should be implemented by dependents. It doesn't belong in the library itself. Can I remove it?

Reply:
Ah hold on, couldn't we simply use `mem::replace`?

Reply:
Well no we can't, sorry. Please use `drop_in_place`.